src/models/state/mod.rs

```diff
@@ -0,0 +1,127 @@
+pub mod state_naive;
+pub mod state_v01;
+pub use state_naive::StateNaive;
+pub use state_v01::StateV01;
+
+use std::fmt::{Display, Formatter, Result as FmtResult};
+
+use serde::de::{Deserialize, Deserializer, Error as DeserializeError};
+use serde::ser::{Error as SerializeError, Serialize, Serializer};
+use serde_derive::{Deserialize, Serialize};
+use strum::IntoEnumIterator;
+use strum_macros::EnumIter;
+
+use super::{Convert, LinkMetadata, PredicateLayout};
+use crate::Error;
+use crate::Result;
+
+#[derive(Debug, Hash, Eq, PartialEq, Clone, Copy, EnumIter)]
+pub enum StateVersion {
```

Comment:
More naming nits here. Let's redo this to reflect `Statement` rather than `State` throughout. "State" is an overloaded CS term as it is and we don't want any confusions.